custom_components/spotcast/__init__.py

```diff
@@ -248,10 +281,25 @@ def start_casting(call: ha_core.ServiceCall):
                 ignore_fully_played,
             )
         else:
-
+            searchResults = []
             if is_empty_str(uri):
                 # get uri from search request
-                uri = get_search_results(search, client, country)
+                searchResults = get_search_results(
+                    spotify_client=client,
+                    limit=limit,
+                    artistName=artistName,
+                    country=country,
+                    albumName=albumName,
+                    playlistName=playlistName,
+                    trackName=trackName,
+                    showName=showName,
+                    episodeName=episodeName,
+                    audiobookName=audiobookName,
+                    genreName=genreName,
+                )
```

Comment:
some people may find this code style unnecessarily verbose - the reason I like it is so that I can mix up positional arguments in any order and it will not cause a bug.